src/BodyParams/BodyParamsMiddleware.php

```diff
@@ -82,10 +83,17 @@ public function __invoke(ServerRequestInterface $request, ResponseInterface $res
             }
 
             // Matched! Parse and pass on to the next
-            return $next(
-                $strategy->parse($request),
-                $response
-            );
+            try {
+                return $next(
+                    $strategy->parse($request),
+                    $response
+                );
+            } catch (MalformedRequestBodyException $exception) {
```

Comment:
My inclination is that we return the 400 response. If a user wants to customize how this works (e.g., to provide logging, or to provide a custom error message, etc.), they can extend the `BodyParamsMiddleware` and/or provide a substitute implementation in their pipeline. In the meantime, we "do the right thing" for the 80% use case.